src/app.module.ts

```diff
@@ -42,6 +43,10 @@ import { LoggerModule } from "./loggers/logger.module";
     ConfigModule.forRoot({
       load: [configuration],
     }),
+    ConditionalModule.registerWhen(
+      MetricsModule,
+      (env: NodeJS.ProcessEnv) => env.METRICS_ENABLED === "yes",
+    ),
```

Comment:
Can we use the configuration module that is imported on line 43?

Reply:
I have checked official documentation and some other articles. `ConditionalModule.registerWhen` method do not support access to the configService. 
The cleanest way is to use `ConditionalModule.registerWhen` and fetch env directly, however, there's hack to achieve same result with DyncmiModule with increased complexity.

Reply:
<img width="902" alt="image" src="https://github.com/user-attachments/assets/ef457026-09a1-4d1e-995b-06eda9570623" />